app/src/main/java/com/alphawallet/app/ui/widget/holder/WalletSummaryHolder.java

```diff
@@ -211,15 +256,25 @@ private Wallet fetchWallet(Wallet w)
         RealmWalletData realmWallet = realm.where(RealmWalletData.class)
                 .equalTo("address", w.address)
                 .findFirst();
-
         if (realmWallet != null)
         {
             w.balance = realmWallet.getBalance();
             w.ENSname = realmWallet.getENSName();
             w.name = realmWallet.getName();
             w.ENSAvatar = realmWallet.getENSAvatar();
-        }
+            if (w.tokens != null)
+            {
+                for (Token t : w.tokens)
+                {
+                    if (t.getWallet().equals(w.address))
+                    {
+                        testNetHorizontalListAdapter = new TestNetHorizontalListAdapter(w.tokens, context);
+                        recyclerView.setAdapter(testNetHorizontalListAdapter);
```

Comment:
Shall we break the loop once matched?

Reply:
@seabornlee ,

I tried breaking the loop once the condition matched but it impacts the display time in the adapter. When we are not breaking a loop at that time the user experience is a bit smooth as compared to when we are breaking a loop. I can share you a video if needed. 

Please share your thoughts.

Reply:
Hi @keval-finimble there should not be any changes to WalletSummaryHolder. This file should be unchanged. We don't need to know the total testnet held - only find specific testnet balance on each wallet.

Reply:
Hi @keval-finimble can you move this handling out into the widget:

Send the w.tokens to your adapter and do all the handling there. Otherwise we have this several layers deep nested bracket; it makes the code more maintainable.